frontend/public/components/utils/datetime.ts

```diff
@@ -114,3 +114,39 @@ export const formatDuration = (ms: number) => {
 
   return formatted;
 };
+
+// Conversions between units and milliseconds
+const s = 1000;
+const m = s * 60;
+const h = m * 60;
+const d = h * 24;
+const w = d * 7;
+const units = {w, d, h, m, s};
+
+// Formats a duration in milliseconds like "1h 10m"
+export const formatPrometheusDuration = (ms: number) => {
```

Comment:
Wondering if we should update `formatDuration` to be this and use it everywhere. Is the primary difference to support longer durations like days and weeks?